File: MORE CODING PLATFORMS/Codeforces/vanyaAndTheLanterns/vanyaAndTheLanterns.py
# 0 3 5 7 9 14 15



from operator import index


arr = input().split()
n = int(arr[0])
l = int(arr[1])
indexes = [int(i) for i in input().split()]
indexes.sort()
diff = max(indexes[0], l-indexes[-1])
for i in range(len(indexes)-1):
    a = indexes[i+1] - indexes[i]
    if a > diff:
        diff = a
print("{:.10f}".format(diff/2))

File: MORE CODING PLATFORMS/Codeforces/vanyaAndTheLanterns/vanyaAndTheLanterns.cpp
#include <stdio.h>
#include <algorithm>
using namespace std;

int n, l, i, indexes[1002], diff;

int main ()
{
    scanf("%d%d", &n, &l);
    for(i = 0; i < n; i++){
        scanf("%d", &indexes[i]);
    }
    sort(indexes, indexes+n);
    diff = 2*max(indexes[0], l - indexes[n-1]);
    for(i = 0; i < n -1; i++){
        diff = max(diff, indexes[i+1]- indexes[i]);
    }
    printf("%.10f\n", diff/2.);
    return 0;
}  